libs/design/media-gallery/src/media-gallery/media-gallery.component.ts

```diff
@@ -37,30 +39,142 @@ let uniqueGalleryId = 0;
     },
   ],
   imports: [
-    DaffMediaRendererComponent,
     DaffThumbnailDirective,
+    NgTemplateOutlet,
   ],
 })
-export class DaffMediaGalleryComponent implements DaffMediaGalleryRegistration, OnInit, OnDestroy {
+export class DaffMediaGalleryComponent implements DaffMediaGalleryRegistration, AfterContentInit {
+  private _id: string;
+
+  /**
+   * @docs-private
+   */
+  @HostBinding('attr.role')
+  role = 'tablist';
+
+  /**
+   * The id of the gallery.
+   */
+  @HostBinding('attr.id')
+  @Input()
+  get id() {
+    return this._id;
+  }
+  set id(val: string | undefined | null) {
+    if(!val){
+      return;
+    }
+    this._id = val;
+  };
+
   /**
    * Adds a class for styling the media gallery
+   * @docs-private
    */
   @HostBinding('class.daff-media-gallery') class = true;
 
   /**
-   * The name of the gallery
+   * The name of the gallery.
+   * @deprecated use `id` instead.
+   */
+  @Input() name: string;
+
+  /**
+   * @docs-private
+   */
+  @ContentChildren(DaffThumbnailDirective) _thumbnails: QueryList<DaffThumbnailDirective>;
+
+  /**
+   * @docs-private
    */
-  @Input() name = `${uniqueGalleryId}`;
+  @ViewChildren('thumbnailButtons', { read: ElementRef }) private _thumbnailButtons: QueryList<ElementRef<HTMLElement>>;
 
-  constructor(private registry: DaffMediaGalleryRegistry) {
+  constructor(private skeletonDirective: DaffSkeletonableDirective) {
     uniqueGalleryId++;
+    this.name = 'gallery-' + uniqueGalleryId;
+  }
+
+  get skeleton() {
+    return this.skeletonDirective.skeleton;
+  }
+
+  /**
+   * @docs-private
+   */
+  _selectedThumbnail: DaffThumbnailDirective = undefined;
+
+  /**
+   * @docs-private
+   */
+  private _selectedIndex: number = undefined;
+
+  private focusSelected() {
+    this._thumbnailButtons.get(this._selectedIndex)?.nativeElement.focus();
   }
 
-  ngOnInit() {
-    this.registry.add(this);
+  /**
+   * Select a specific entry in the media gallery by its index (starting at 0).
+   */
+  selectIndex(index: number) {
+    if(this._thumbnails.get(index)){
+      this._selectedIndex = index;
+      this._selectedThumbnail?.deselect();
+      this._thumbnails.get(index).select();
+      this._selectedThumbnail = this._thumbnails.get(index);
+    }
+  }
+
+  /**
+   * Select a specific thumbnail for this gallery.
+   */
+  selectThumbnail(thumbnail: DaffThumbnailDirective) {
```

Comment:
What is the use case of allowing users to choose a specific thumbnail to be selected by default? Shouldn't it always default to the first thumbnail?

Reply:
This function is no longer in the code.